consensus/misc/eip1559/eip1559.go

```diff
@@ -58,6 +58,12 @@ func VerifyEIP1559Header(config *params.ChainConfig, parent, header *types.Heade
 // CalcBaseFee calculates the basefee of the header.
 // The time belongs to the new block to check if Canyon is activted or not
 func CalcBaseFee(config *params.ChainConfig, parent *types.Header, time uint64) *big.Int {
+	// If this is the cel2 transition block then we use the initial base fee
+	// from the Cel2Config of the chain config.
+	if config.Cel2Time != nil && *config.Cel2Time == time {
+		return new(big.Int).SetUint64(config.Cel2Config.TransitionBlockBaseFee)
```

Comment:
Do we need this to be configurable? Can't we just hardcode it (there will be no txs in the block anyways) or use the basefee of the block before?

Reply:
Hardcoding is what ethereum did with `InitialBaseFee` but we probably don't want to change that, and it doesn't match up with our current base fees. Just taking the base fee from the block works for migrating post gingerbread blocks, but not for pre gingerbread. So we could do away with this, but we'd lose the ability to migrate pre-gingerbread blocks.

Reply:
We'll probably never migrate pre-gingerbread blocks, so I'd avoid the complexity. But no hard opinion on my side.

Reply:
It's useful if you want to test locally with a small portion of one of our established networks, but that's it.

So I think I'm going to go for just taking the parent base fee value, and I'll update the migration script to fail if an attempt is made to migrate at a pre-gingerbread block.

Reply:
Or just use some reasonable default in that case?

Reply:
Hi @palango updated the approach to choosing the base fee here.